index.js

```diff
@@ -9,6 +9,7 @@ const allRules = {
   'button-has-type': require('./lib/rules/button-has-type'),
   'default-props-match-prop-types': require('./lib/rules/default-props-match-prop-types'),
   'destructuring-assignment': require('./lib/rules/destructuring-assignment'),
+  'declarations-for-function-components': require('./lib/rules/declarations-for-function-components'),
```

Comment:
let's keep the PR as a draft until we settle on a name :-)

Reply:
Something not specific to the different types of function definitions maybe. Maybe something like `standard-function-component-definition`. The function definition type can be specified in the options then with default being arrow function (since it's the most popular I think).